### PR TITLE
[Draft] Re-enable and update version of ghc

### PR DESCRIPTION
**Pull request by danwdart**
Maybe this will somewhat help the build be compatible with the latest llvm, I'm not sure.

### CODE REVIEW

File: disabled-packages/ghc/build.sh [hunk 1]
```diff
@@ -3,10 +3,10 @@
 TERMUX_PKG_HOMEPAGE=https://www.haskell.org/ghc/
 TERMUX_PKG_DESCRIPTION="The Glasgow Haskell Compilation system"
 TERMUX_PKG_LICENSE="BSD 2-Clause, BSD 3-Clause, LGPL-2.1"
-TERMUX_PKG_VERSION=8.10.1
-TERMUX_PKG_REVISION=5
+TERMUX_PKG_VERSION=9.2.1
+TERMUX_PKG_REVISION=6
```
Comment (thunder-coding) [inline disabled-packages/ghc/build.sh:7]: You should remove `TERMUX_PKG_REVISION` as version has been updated
```suggestion
```

**Review by thunder-coding — CHANGES_REQUESTED** [review]

File: disabled-packages/ghc/build.sh (hunk 1 above)
Comment (thunder-coding) [reply disabled-packages/ghc/build.sh:7]: Also move this form `disabled-packages/` to `packages/`

Comment (danwdart) [reply disabled-packages/ghc/build.sh:7]: Ah fair enough. Wasn't sure it was getting used.

Comment (danwdart) [reply disabled-packages/ghc/build.sh]: I don't know about the fix - need to try that out.

**Comment by xeffyr** [comment]
You need to move `ghc` directory to `/packages` so it will no longer be disabled package and will be picked by GitHub Actions. Additionally you may need to update https://github.com/termux/termux-packages/blob/master/scripts/build/setup/termux_setup_ghc.sh.

GHC should support at least Clang `12.0.8` which is currently used by Android NDK. Downgrading isn't possible.

**Comment by leapofazzam123** [comment]
> You need to move `ghc` directory to `/packages` so it will no longer be disabled package and will be picked by GitHub Actions. Additionally you may need to update https://github.com/termux/termux-packages/blob/master/scripts/build/setup/termux_setup_ghc.sh.
> 
> GHC should support at least Clang `12.0.8` which is currently used by Android NDK. Downgrading isn't possible.

@danwdart

**Comment by leapofazzam123** [comment]
If you don't move this from disabled_packages, it's still marked as a disabled packagw

**Comment by danwdart** [comment]
Yes, understood, will integrate on local - don't know if there's anything special I need to do locally to test the build.

**Pull request closed without merging**